pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/MethodTypeResolution.java

```diff
@@ -510,4 +510,12 @@ public static boolean isSubtypeable(JavaTypeDefinition parameter, JavaTypeDefini
 
         return false;
     }
+
+    public static JavaTypeDefinition boxPrimitive(JavaTypeDefinition def) {
+        if (!def.isPrimitive()) {
+            return null;
+        }
+
+        return JavaTypeDefinition.forClass(BOXED_PRIMITIVE_SUBTYPE_ORDER.get(PRIMITIVE_SUBTYPE_ORDER.indexOf(def.getType())));
```

Comment:
Then a bi-map would be needed. I'll keep thinking about it, for the time being let's just move on...